pro/src/components/firebase/FirebaseContainer.js

```diff
@@ -0,0 +1,14 @@
+import { useEffect } from 'react'
+import { useLocation } from 'react-router-dom'
+
+import { analytics } from 'utils/firebase'
+
+const FirebaseAnalytics = () => {
+  let location = useLocation()
+  useEffect(() => {
+    analytics.logPageView(location.pathname)
```

Comment:
Cela fonctionne bien sur toutes les pages que j'ai testée sauf `/offres` où l’évènement est trigger 3 fois. Je ne sais pas trop comment être sur qu'on trigger qu'une fois l'évènement

Reply:
on pourrait avoir un listener sur `history` et verifier que next et current url on des valeurs differents:

https://stackoverflow.com/questions/41911309/how-to-listen-to-route-changes-in-react-router-v4
```javascript
import { withRouter } from 'react-router-dom';

    const myComponent = ({ history }) => {

        history.listen((location, action) => {
            // location is an object like window.location
            console.log(action, location.pathname, location.state)
        });

        return <div>...</div>;
    };

    export default withRouter(myComponent);
```

Reply:
Aussi j'allais oublié mais tout les nouveau fichiers sont codé en Typescript

Reply:
note: avec `hisory.listen` on pourrait :
* avoir une information "go to" ou "go back to" selon que l'action est push ou pop.
* avoir un information "reload with parameters" par example si l'utilisateur passe de `/offres/` à `/offres/?monFiltre=maValeur`